launcher/minecraft/PackProfile.cpp

```diff
@@ -939,6 +945,32 @@ bool PackProfile::installCustomJar_internal(QString filepath)
     return true;
 }
 
+bool PackProfile::installComponents_internal(QStringList filepaths)
+{
+    const QString patchDir = FS::PathCombine(d->m_instance->instanceRoot(), "patches");
+    if (!FS::ensureFolderPathExists(patchDir))
+        return false;
+
+    for (const QString& source : filepaths) {
+        const QFileInfo sourceInfo(source);
+
+        auto versionFile = ProfileUtils::parseJsonFile(sourceInfo, false);
```

Comment:
I haven't tested how this method handles invalid files. From a quick sweep over the code it looked like it shouldn't cause issues for us